apps/spruce/src/pages/waterfall/PaginationButtons/index.tsx

```diff
@@ -0,0 +1,64 @@
+import styled from "@emotion/styled";
+import Button from "@leafygreen-ui/button";
+import { useWaterfallAnalytics } from "analytics";
+import Icon from "components/Icon";
+import { size } from "constants/tokens";
+import { WaterfallPagination } from "gql/generated/types";
+import { useQueryParams } from "hooks/useQueryParam";
+import { WaterfallFilterOptions } from "types/waterfall";
+
+interface PaginationButtonsProps {
+  pagination: WaterfallPagination | undefined;
+}
+
+export const PaginationButtons: React.FC<PaginationButtonsProps> = ({
+  pagination,
+}) => {
+  const { hasNextPage, hasPrevPage, nextPageOrder, prevPageOrder } =
+    pagination ?? {};
+  const { sendEvent } = useWaterfallAnalytics();
+  const [queryParams, setQueryParams] = useQueryParams();
+
+  const onNextClick = () => {
+    sendEvent({ name: "Changed page", direction: "next" });
+    setQueryParams({
+      ...queryParams,
+      [WaterfallFilterOptions.MaxOrder]: nextPageOrder,
+      [WaterfallFilterOptions.MinOrder]: undefined,
+    });
+  };
+
+  const onPrevClick = () => {
+    sendEvent({
+      name: "Changed page",
+      direction: "previous",
+    });
+    setQueryParams({
+      ...queryParams,
+      [WaterfallFilterOptions.MaxOrder]: undefined,
+      [WaterfallFilterOptions.MinOrder]: prevPageOrder,
+    });
+  };
```

Comment:
Could we implement these with [useTransition](https://react.dev/reference/react/useTransition)? It would improve the ability to interact with the page while cached data is rendering as described above. (I haven't tried it myself so hopefully this doesn't break the loading animation for non-cached pages)

Reply:
Tried it (hopefully correctly) and pushed it, but it's looking like the loading animations did indeed disappear...